pkg/scheduler/ugm/manager.go

```diff
@@ -365,127 +381,159 @@ func (m *Manager) internalProcessConfig(cur configs.QueueConfig, queuePath strin
-				ut := m.userTrackers[user]
-				ut.setGroupForApp(app, nil)
+func (m *Manager) clearEarlierSetUserLimits(newUserLimits map[string]map[string]*LimitConfig) {
+	if len(m.userLimits) == 0 {
+		return
+	}
+	m.RLock()
+	defer m.RUnlock()
+	// Need to clear user config only when
+	// 1. config set earlier but not now
+	// 2. user already tracked for the queue path
+	for queuePath, limitConfig := range m.userLimits {
+		// Is queue path exists?
+		if newUserLimit, ok := newUserLimits[queuePath]; !ok {
+			for u := range limitConfig {
+				if ut, utExists := m.userTrackers[u]; utExists {
+					m.clearEarlierSetUserLimit(ut, queuePath)
+				}
+			}
+		} else {
+			// Queue path exists. Is user exists?
+			for u := range limitConfig {
+				if _, ulExists := newUserLimit[u]; !ulExists {
+					if ut, utExists := m.userTrackers[u]; utExists {
+						m.clearEarlierSetUserLimit(ut, queuePath)
+					}
+				}
 			}
 		}
 	}
-
-	// Clear already configured limits of user for which limits have been configured before but not now
-	for _, ut := range m.userTrackers {
-		m.clearEarlierSetUserLimits(ut, queuePath, userLimits)
-	}
-	return nil
 }
 
-func (m *Manager) clearEarlierSetUserLimits(ut *UserTracker, queuePath string, userLimits map[string]bool) {
+func (m *Manager) clearEarlierSetUserLimit(ut *UserTracker, queuePath string) {
 	// Is this user already tracked for the queue path?
 	if ut.IsQueuePathTrackedCompletely(queuePath) {
-		u := ut.userName
-		// Is there any limit config set for user in the current configuration? If not, then clear those old limit settings
-		if _, ok := userLimits[u]; !ok {
-			log.Log(log.SchedUGM).Debug("Need to clear earlier set configs for user",
-				zap.String("user", u),
-				zap.String("queue path", queuePath))
-			// Is there any running applications in end queue of this queue path? If not, then remove the linkage between end queue and its immediate parent
-			if ut.IsUnlinkRequired(queuePath) {
-				ut.UnlinkQT(queuePath)
-			} else {
-				ut.setLimits(queuePath, resources.NewResource(), 0)
-				log.Log(log.SchedUGM).Debug("Cleared earlier set limit configs for user",
-					zap.String("user", u),
-					zap.String("queue path", queuePath))
+		log.Log(log.SchedUGM).Debug("Need to clear earlier set configs for user",
+			zap.String("user", ut.userName),
+			zap.String("queue path", queuePath))
+		ut.setLimits(queuePath, resources.NewResource(), 0)
+		// Is there any running applications in end queue of this queue path? If not, then remove the linkage between end queue and its immediate parent
+		if ut.IsUnlinkRequired(queuePath) {
+			ut.UnlinkQT(queuePath)
+		}
+		log.Log(log.SchedUGM).Debug("Cleared earlier set limit configs for user",
+			zap.String("user", ut.userName),
+			zap.String("queue path", queuePath))
+		if ut.canBeRemoved() {
+			delete(m.userTrackers, ut.userName)
+		}
+	}
+}
+
+func (m *Manager) clearEarlierSetGroupLimits(newGroupLimits map[string]map[string]*LimitConfig) {
```

Comment:
Similar comment as for the user: comments for the functions to explain what they do and the names look too much alike